force-app/main/default/classes/AccountTriggerHandler.cls

```diff
@@ -0,0 +1,72 @@
+public with sharing class AccountTriggerHandler {
+    public static void setTypeProspectAfterInsert(List<Account> newAccounts) {
+        List<Account> accountsToUpdate = new List<Account>();
+        //System.debug('number of new accounts: ' + newAccounts.size());
+        for (Account acc : newAccounts) {
+            if (acc.Type == null) {
+                Account accToUpdateAfterInsert = new Account(Id = acc.Id , Type = 'Prospect');
+                accountsToUpdate.add(accToUpdateAfterInsert);
+                //System.debug('Added to update list: ' + accToUpdateAfterInsert);
+            }
+        }
+        //System.debug('number of accounts being updated with prospect: ' + accountsToUpdate.size());
+
+        if (!accountsToUpdate.isEmpty()) {
+            update as system accountsToUpdate;
+        }
+    }
+
+    public static void addressCopyAfterInsert(List<Account> newAccounts) {
+        List<Account> accountsToUpdate = new List<Account>();
+        for (Account acc : newAccounts) {
+            if (acc.ShippingStreet != null ||
+                acc.ShippingCity != null ||
+                acc.ShippingState != null ||
+                acc.ShippingPostalCode != null ||
+                acc.ShippingCountry != null) {
+                
+                Account accToUpdate = new Account(
+                    Id = acc.Id,
+                    BillingStreet = acc.ShippingStreet,
+                    BillingCity = acc.ShippingCity,
+                    BillingState = acc.ShippingState,
+                    BillingPostalCode = acc.ShippingPostalCode,
+                    BillingCountry = acc.ShippingCountry
+                );
+                accountsToUpdate.add(accToUpdate);
+            }
+        }
+        update as system accountsToUpdate;
+    }
+
+    public static void setRating(List<Account> newAccounts) {
+        List<Account> accountsToUpdate = new List<Account>();
+        for (Account acc : newAccounts) {
+            if (acc.Phone != null && acc.Website != null && acc.Fax != null) {
+
+                Account accToUpdate = new Account(
```

Comment:
You don't need to create a new Account here. You can update the account you are looping over.